src/kloewe/dot/src/dot.h

```diff
@@ -0,0 +1,92 @@
+/*----------------------------------------------------------------------------
+  File    : dot.h
+  Contents: dot product (cpu dispatcher)
+  Author  : Kristian Loewe
+----------------------------------------------------------------------------*/
+#ifndef DOT_H
+#define DOT_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*----------------------------------------------------------------------------
+  Enum to encode the sets of dot implementations
+----------------------------------------------------------------------------*/
+enum flags {
+    DOT_NAIVE  = 1,
+    DOT_SSE2   = 2,
+    DOT_AVX    = 3,
+    #ifndef DOT_NOFMA
+    DOT_AVXFMA = 4,
+    #endif
+    DOT_AUTO   = 3
```

Comment:
Actually, scratch that, how about `DOT_AUTO = 100`, so it won't need to be changed any time soon.  The idea is to have AUTO select AVXFMA when available, right?